### ISSUE TITLE
Replace martial arts books with dojo NPCs

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#41359]
# Is your feature request related to a problem? Please describe.
Learning martial arts from a book is an elderly feature from the days before NPCs could teach you martial arts. We have had the ability to get martial arts from NPCs for a while, but rarely use it.

# Describe the solution you'd like
1. Remove most martial arts books, and change the remaining ones to unarmed training manuals.
2. Using nested mapgen, give existing dojos a chance to spawn an NPC from a list of common and semi-common martial arts trainers.
3. Add rare wilderness dojos that spawn NPCs with a chance of teaching rarer martial arts.
4. Add extremely rare martial arts trainer NPCs in special locations, such as as a prisoner in the mi-go base, or in a secret room inside the mines. A chance of travelling NPC martial arts trainers (dynamic npcs) is also reasonable, but that would then be so rare as to be almost nonexistent

Dojo NPCs should have various costs or requirements to train you. For example, an NPC might require you to have unarmed combat 4, or to defeat their last student (who is now, coincidentally, a zombie), or to find them a good quality katana.

Dojo NPCs should avoid all being the "Mr Miyagi" stereotype (that should, obviously, be the karate trainer NPC, but only that one). However it would be reasonable for them to be pretty silly characters, since this is a pretty silly feature.

Dojo NPCs should probably have some kind of cheat mutation that gives them excellent dodging ability and high hand to hand damage, since NPCs can't currently use martial arts properly AFAIK.

# Describe alternatives you've considered
Adding in hard coded trainers for each martial art with an associated quest line and faction would be lovely, but we have a lot of martial arts.

**Comment by Yunatan** [comment]
I like the motivation behind this but I just want to point out an issue I see with this. Right now martial arts are already so plentiful yet so different that finding a book with a desired martial art can be a challenge, even considering that every dojo spawn several books and a big city can potentially hold several dojos. If every dojo would contain just one NPC (not even guaranteed) and that NPC would teach you only one martial art, finding the one you want your character built around would be a massive pain. And it will only worsen with the expansion of martial arts. 
I think a really cool solution to that would be a feature where a Dojo NPC can give you directions (like on the map) to other Dojo NPCs, maybe even in regards to particular martial art you want. That way a search for desirable martial art can turn into a semi-quest which is fun and not chore-like.

**Comment by I-am-Erk** [comment]
I think specifics like dojos redirecting you to other NPC's, and quests that lead you to other trainers, are great ideas and should be picked up by whomever decides to follow up on this. Don't forget though that this suggestion does include adding more dojos and other places to get trained... If things like triffid groves or mi-go bases have a chance to spawn trainers you can rescue, for example, that gives them another interesting reason to raid, *and* increases martial arts spawns

**Comment by meladath** [comment]
This really breaks my immersion.... we have NPC's that magically know where other trainers are and what they can teach? And they all somehow survived the cataclysm? I agree martial arts needs a rework but... this really seems like the wrong way to go about it imo.

**Comment by Zireael07** [comment]
Why do you think a teacher NPC would be insanely OP?

**Comment by meladath** [comment]
> 
> 
> Why do you think a teacher NPC would be insanely OP?

I think this becausae in the OP it states "Dojo NPCs should probably have some kind of cheat mutation that gives them excellent dodging ability and high hand to hand damage, since NPCs can't currently use martial arts properly AFAIK."

So in theory they will be killing machines.

**Comment by I-am-Erk** [comment]
You need to learn some shades of grey. "Excellent dodging ability and hand to hand damage" is not "killing machine". Read the context. The objective would be to make them equivalent to a player using martial arts.

Finding npcs that can train you in martial arts doesn't mean that "they all survived the apocalypse", the recommendation isn't to make every martial arts instructor in New England available to teach you. I'm not sure how this is more immersion breaking to you than readily available books that instantaneously teach you krav maga

**Comment by mlangsdorf** [comment]
It wouldn't be super hard to make NPCs use martial arts.  They can already use techniques, they just can't select martial arts yet.  70% of the work would be annotating the existing martial arts with an NPC_FRIENDLY tag, because even if NPCs were able to select Niten-Ichi Ryu, they would die if they tried to actually use it.

Rather than giving them a bonus dodge and hand to hand damage, we might give instructors for non-NPC_FRIENDLY arts a secret, hidden, unteachable martial art with a bunch of static bonuses that resembles their martial art.

**Comment by esotericist** [comment]
> This would also remove any possibility of learning martial arts on any characters that have no NPCs in their world which is pretty game breaking if you ask me.

We are under no obligation to keep the game playable if people want to excise part of the core design from it. Needing NPCs for things is going to continue to happen.

**Comment by meladath** [comment]
> 
> 
> > This would also remove any possibility of learning martial arts on any characters that have no NPCs in their world which is pretty game breaking if you ask me.
> 
> We are under no obligation to keep the game fully playable if people want to excise part of the core design from it. Needing NPCs for things is going to continue to happen.

So anyone who plays with a no NPC's (a huge portion of starts) start simply can never learn martial arts?

**Comment by esotericist** [comment]
> So anyone who plays with no starting NPC's (a huge portion of proffs/scenarios...) simply can never learn martial arts unless they pick it at start? Can you give another example of a gameplay mechanic associated completely with the player (i.e. not related to NPCs) that is locked behind requiring NPCs?

The team has been fairly open about intent to make more parts of the game absolutely require NPCs. This includes a lot of crafting tasks, learning certain proficiencies (and martial arts can be argued to fall in this umbrella), installing bionics, and probably other things I don't recall.

Just because it isn't like that *yet* doesn't mean that isn't the plan.

> Learning martial arts from books isn't immersion breaking at all considering how you learn other skills in this game. 

How you learn other skills in the game *is* a problem, one that is intended to be corrected.

**Comment by meladath** [comment]
>How you learn other skills in the game is a problem, one that is intended to be corrected.

Really? Damn, I'll have to stop playing when that changes then. What a shame.

>edit: I just noticed you specified 'starting NPCs'. That's irrelevant. we're talking about other NPCs besides the ones that start where your character is, which are NOT affected by profession/scenario

I thought if you chose a start that had no "starting NPCs" then there were no NPCs at all in the world? Or was that an old bug that I haven't noticed has been fixed yet?

**Comment by anothersimulacrum** [comment]
> I thought if you chose a start that had no "starting NPCs" then there were no NPCs at all in the world? Or was that a really old bug I hadn't noticed got fixed? Or simply a wrong assumption? I could just be completely wrong.

I don't know of that being a bug, but I don't know everything that ever was a bug. Starting NPCs has no impact on NPCs in the world.

**Comment by esotericist** [comment]
my guess is that there's conflation between 'static NPCs' (the global setting for static NPCs spawning in the world) and the starting NPC setting (which could be always, scenario-specific, or never, and was dependant on the previous setting.

**Comment by meladath** [comment]
I think it may just be an old thing I am remembering wrong. I have been inactive, especially in the NPC regard... since 0.C  days.

**Comment by I-am-Erk** [comment]
I'd appreciate it if you'd make your comments more carefully instead of editing them constantly over hours. It's exhausting to keep up with.

**Comment by meladath** [comment]
Sorry, next time I'll leave in all my typos and gramattical errors :)

**Comment by anothersimulacrum** [comment]
You're significantly altering the meaning and content of your messages, not just fixing typos.

**Comment by meladath** [comment]
And won't edit any of my messages, posting another comment is just easier I guess.

So? Am I not allowed to? No one has replied to my messages. I thought I was allowed to edit something I have written before anyone has even replied. My apologies.

**Comment by meladath** [comment]
my comments* whoops, typo. Damn it, sorry....

**Comment by I-am-Erk** [comment]
You changed 

> > How you learn other skills in the game is a problem, one that is intended to be corrected.
> 
> Really? Damn, I'll have to stop playing when that changes then. What a shame.
> 
> > edit: I just noticed you specified 'starting NPCs'. That's irrelevant. we're talking about other NPCs besides the ones that start where your character is, which are NOT affected by profession/scenario
>
> I thought if you chose a start that had no "starting NPCs" then there were no NPCs at all in the world? Or was that an old bug that I haven't noticed has been fixed yet?

to

> > The team has been fairly open about intent to make more parts of the game absolutely require NPCs.
>
> They have? Thank gods this game its open source so I can locally revert those commits when its pushed. Anything mandatory involving NPCs is a huge detraction for my personal enjoyment of the game. But thats the joy of open source.
> 
> > How you learn other skills in the game is a problem, one that is intended to be corrected.
>
> Really? That's odd, I was under the assumption that it was a core gameplay mechanic, as in, you were pretty much supposed to learn skills via books, not by making 40 pairs of arm sleeves and cutting them up between each craft. No biggie, can be reverted locally easily enough. Maybe I can mod back in the old way if enough people want it. Shouldn't be too hard at all.
>
> > edit: I just noticed you specified 'starting NPCs'. That's irrelevant. we're talking about other NPCs besides the ones that start where your character is, which are NOT affected by profession/scenario
>
> I thought if you chose a start that had no "starting NPCs" then there were no NPCs at all in the world? Or was that a really old bug I hadn't noticed got fixed? Or simply a wrong assumption? I could just be completely wrong.

over several hours. That's adding paragraphs and doubling the size of texts. Fix typos as needed but don't double the size of the post.

**Comment by meladath** [comment]
> 
> 
> You changed
> 
> > > How you learn other skills in the game is a problem, one that is intended to be corrected.
> > 
> > 
> > Really? Damn, I'll have to stop playing when that changes then. What a shame.
> > > edit: I just noticed you specified 'starting NPCs'. That's irrelevant. we're talking about other NPCs besides the ones that start where your character is, which are NOT affected by profession/scenario
> > 
> > 
> > I thought if you chose a start that had no "starting NPCs" then there were no NPCs at all in the world? Or was that an old bug that I haven't noticed has been fixed yet?
> 
> to
> 
> > > The team has been fairly open about intent to make more parts of the game absolutely require NPCs.
> > 
> > 
> > They have? Thank gods this game its open source so I can locally revert those commits when its pushed. Anything mandatory involving NPCs is a huge detraction for my personal enjoyment of the game. But thats the joy of open source.
> > > How you learn other skills in the game is a problem, one that is intended to be corrected.
> > 
> > 
> > Really? That's odd, I was under the assumption that it was a core gameplay mechanic, as in, you were pretty much supposed to learn skills via books, not by making 40 pairs of arm sleeves and cutting them up between each craft. No biggie, can be reverted locally easily enough. Maybe I can mod back in the old way if enough people want it. Shouldn't be too hard at all.
> > > edit: I just noticed you specified 'starting NPCs'. That's irrelevant. we're talking about other NPCs besides the ones that start where your character is, which are NOT affected by profession/scenario
> > 
> > 
> > I thought if you chose a start that had no "starting NPCs" then there were no NPCs at all in the world? Or was that a really old bug I hadn't noticed got fixed? Or simply a wrong assumption? I could just be completely wrong.
> 
> over several hours. That's adding paragraphs and doubling the length of text. Fix typos as needed but don't double the size of the post.

Thanks for pointing this out.
I didn't know, thought I was just editing it for fun, not trying to change the point of my message so people would easily understand my point.

Oh wait, I was editing it to better get my message across. Sorry, it's ok though, like I said, next time I'll just post another comment before anyone has replied instead, as that is clearly your favoured option.

**Comment by I-am-Erk** [comment]
My preferred option is that you think about what you're going to say before you click "comment", rather than constantly realizing you had more you wanted to add and editing it back in.

**Comment by meladath** [comment]
Yes, I will make sure I think much more before commenting then editing it.

**Comment by mlangsdorf** [comment]
As far as the game being open source and you have the option to revert commits that you don't like, that's absolutely true and no one can or wants to take that ability away from you.

On the other hand, this the website for the CleverRaven version of C:DDA, and necessarily the focus is going to be on making changes that the extended CleverRaven team wants to make.  You are welcome to polite discuss your disagreements with those changes, but please lay off the snark.  We already know that not all of the changes we are proposing are going to be popular and some people will want to revert them.  It isn't necessary to remind us of that, and it's not a persuasive argument for or against adopting some changes.

FWIW, while I support adding NPCs to teach martial arts, I think we should leave martial arts books in the game.  Martial arts instruction manuals are a real world thing, and with sufficient pre-existing skill and sufficient practice, it should be possible to pick up a style from a book.  It should take much longer, and require higher skills, than learning by instruction.

**Comment by I-am-Erk** [comment]
With regard to your concerns, I don't think you have a very clear understanding either of what the suggestion here is, nor of what the game direction is.

- NPCs training you in martial arts need to be capable of, roughly, those martial arts. Whether that's through a clunky mutation or through adding martial arts in (which sounds like it's simpler than I thought) is irrelevant, that's the intent. Beyond that there's no reason to interpret anything about their OPedness.
- If NPCs wander out of their dojos and get eaten by zombies, that's a solid representation of their relative scarcity. However, this problem wouldn't apply to the several other martial arts training NPCs I have suggested.
- Learning skills through practice is not relevant to this Issue, but as with other things you've just jumped to an incorrect conclusion about what we mean. I would recommend asking questions about things you don't understand, rather than just assuming the other person hasn't thought of basic problems. [This issue](https://github.com/CleverRaven/Cataclysm-DDA/issues/28283) is now outdated compared to current plans, but has the basic gist of it.

**Comment by esotericist** [comment]
once we have proficiencies as intended, making martial arts manuals have appropriate prerequisites and present the necessary hurdles would be straightforward.

**Comment by meladath** [comment]
Yes I made a lot of bad assumptions and simply spewed rubbish without actually thinking things through properly. My apologies. I have deleted a lot of my stupid comments.

I still think we should have training manuals, but they should much, much longer to learn. Maybe something like a day of reading/training per level (which somehow equates to 1 ability per level or so?).

**Comment by Tharn** [comment]
Putting NPCs into city dojos makes no sense. Cities are overrun. You don't hang out there and "defend your dojo", you become a survivor like everybody else and find a better place to make your home.

I like the idea of special martial arts trainers being in the game. But put them in the refugee center or other pacified communal areas. Hell, put some in log cabins out in the forest, that's fine too. Martial arts experts being hermits is good flavor. Then we could have quests from other survivors to the tune of "I heard there's a shaolin master living out there somewhere between coordinates X and Y."

Just don't put them in cities. I can see it now: The inevitable _race against the clock_ as the player reality bubble touches on a dojo with an NPC and the player sprints across the street to frantically try and lure away or kill all nearby zombies before the NPC gets insta-gibbed. How'd he survive there before? Not realistic.

Also, books are rare and cool. They're one of the things that from early game to late game, players enjoy to hunt for. And there's enough Martial Arts that even with books and NPCs in the game, there's plenty to do.

So I would say yes to 3. and 4. but a resounding no to the first two.

**Comment by I-am-Erk** [comment]
The game has moved a little since 2020, it's probably less necessary to spawn NPCs in the actual dojos as we now have significantly more safe spaces in other areas that they could migrate to. However, unless I missed a change somewhere, the concern about learning martial arts from books as if through the matrix remains a big one. NPCs should still be replacing that extremely jank mechanic. Now that we have proficiencies, if someone wants to preserve martial arts books, we should probably make proficiencies that link to each martial art, and allow a book to train up to a fixed percentage of that martial art, and perhaps activate some practice actions that can take you further, but there's no good argument for preserving "I read a book and now I am great at kung fu".

**Comment by I-am-Erk** [comment]
The current state of martial arts and skill learning is very different, at this point this wouldn't be the solution I'd want so I'm going to reduce issue clutter and close this.
